### DISCUSSION TITLE
Internal server error. httpSever.listen failed

### DISCUSSION THREAD

**Discussion by emanet29** [serverpod/serverpod#2348] [Q&A]
Hi,

I'm currently testing Serverpod on an online server.
It works perfectly but usually after 1 or 2 hours, the server crashes, here are the logs:

**Internal server error. httpSever.listen failed.
SocketException: Connection reset by peer (OS Error: Connection reset by peer, errno = 104), address = ::, port = 8080
Server default stopped**

Do you have any idea where this could come from?
Thanks

**Comment by Isakdl** [comment]
Hey @emanet29 do you have more information. Such as what environment are you hosting in? Do you have any more logs? Which OS are you running on?

Seems like there is an error coming from the operating system that causes the HTTP server to stop.

**Reply by emanet29** [reply]
Thanks for your help

It's a virtual server with Ubuntu 20.04.6 LTS
Dart SDK version: 3.4.0 (stable) (None) on "linux_x64"
Serverpod 2.0.0
I don't have any more relevant information in the same time slot with "journalctl -f"
Some times the error is : 
**SocketException: Broken pipe (OS Error: Broken pipe, errno = 32), address = ::, port = 8080**
What types of logs could help me?

**Comment by emanet29** [comment]
With a reverse proxy, it seems that is better !
